src/librustc_infer/infer/error_reporting/need_type_info.rs

```diff
@@ -88,6 +88,17 @@ impl<'a, 'tcx> Visitor<'tcx> for FindHirNodeVisitor<'a, 'tcx> {
         if let (None, Some(ty)) =
             (self.found_local_pattern, self.node_ty_contains_target(local.hir_id))
         {
+            // There's a trade-off here - we can either check that our target span
```

Comment:
```suggestion
            // FIXME: There's a trade-off here - we can either check that our target span
```
I think that while this is currently already a huge improvement we more prominently acknowledge that this trade-off should still be removed, e.g. one could add a special case for assignments here in the future.